libs/SalesforceAnalytics/src/com/salesforce/androidsdk/analytics/security/DecrypterInputStream.java

```diff
@@ -37,37 +36,23 @@
 import java.util.List;
 import javax.crypto.Cipher;
 import javax.crypto.CipherInputStream;
-import javax.crypto.spec.IvParameterSpec;
-import javax.crypto.spec.SecretKeySpec;
 
 /** Input stream that decrypts content written with a EncrypterOutputStream */
 public class DecrypterInputStream extends InputStream implements WatchableStream {
 
-    private static final String PREFER_CIPHER_TRANSFORMATION = "AES/CBC/PKCS5Padding";
-
     private InputStream cipherInputStream;
     private List<Watcher> watchers;
 
     public DecrypterInputStream(FileInputStream inputStream, String encryptionKey)
             throws GeneralSecurityException, IOException {
-
-        byte[] iv = new byte[16];
+        int ivLength = inputStream.read();
```

Comment:
Writing iv length so that we can handle iv length changes in the future.
The other option would be to pass the expected length as one of the constructor parameter.